frontend/public/components/utils/useEventListener.jsx

```diff
@@ -0,0 +1,26 @@
+import { useRef, useEffect } from 'react';
+
+const useEventListener = (eventName, handler, element = window) => {
+  const updatedHandler = useRef();
+
+  useEffect(() => {
+    updatedHandler.current = handler;
+  }, [handler]);
+
+  useEffect(() => {
+    const isSupported = element && element.addEventListener;
+    if (!isSupported) {
+      return;
+    }
+
+    const eventListener = (event) => updatedHandler.current(event);
+    element.addEventListener(eventName, eventListener, { passive: false });
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

> ***passive***
> A Boolean that, if true, indicates that the function specified by listener will never call preventDefault(). If a passive listener does call preventDefault(), the user agent will do nothing other than generate a console warning.

Do we really want this as the default behavior?